edge-modules/TwinTester/Program.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("TwinTester");
+
+        static async Task Main()
+        {
+            Logger.LogInformation($"Starting twin tester with the following settings:\r\n{Settings.Current}");
+
+            try
+            {
+                RegistryManager registryManager = RegistryManager.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString);
+
+                ModuleClient moduleClient = await ModuleUtil.CreateModuleClientAsync(
+                    Settings.Current.TransportType,
+                    ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                    ModuleUtil.DefaultTransientRetryStrategy,
+                    Logger);
+
+                AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = Settings.Current.AnalyzerUrl };
+
+                Storage storage = new Storage();
+                storage.Init(Settings.Current.StoragePath, new SystemEnvironment(), Settings.Current.StorageOptimizeForPerformance);
+
+                TwinOperator twinOperator = new TwinOperator(registryManager, moduleClient, analyzerClient, storage);
+
+                using (var timers = new Timers())
+                {
+                    // setup the twin update timer
+                    timers.Add(
+                        Settings.Current.TwinUpdateFrequency,
+                        Settings.Current.JitterFactor,
+                        () => twinOperator.PerformUpdates());
+
+                    TimeSpan validationInterval = new TimeSpan(Settings.Current.TwinUpdateFailureThreshold.Ticks / 4);
```

Comment:
Can we define this validation interval better? Checking four times per timespan potentially allows a valid update made in the last few seconds to fail the validation.